src/config.cpp

```diff
@@ -124,6 +118,16 @@ Variables readVariables(const nlohmann::json& json) {
     return variables;
 }
 
+// read variables in manifest section for circuit_config
+Variables readVariables(const nlohmann::json& json) {
+    Variables variables;
+
+    if (json.find("networks") == json.end() || json.find("manifest") == json.end()) {
```

Comment:
@mgeplf : is the check of `networks` section mandatory for CircuitConfig while parsing manifest? Otherwise I could reuse the function`readVariables` for SimulationConfig without splitting it.

Reply:
Yes, it's mandatory according to the specification:
https://bbpteam.epfl.ch/documentation/projects/circuit-documentation/latest/sonata_config.html#networks

Reply:
However, if that's being validated somewhere else, we can rely on that.  We may need to add more test cases to make sure we're not missing anything if we change this.

Reply:
Looks like it's validated ~line 176, so it should be safe to remove from `readVariables`